### PR TITLE
tests: make snap gc more stable

### PR DESCRIPTION
**Pull request by BusyJay**
@overvenus @siddontang PTAL

### CODE REVIEW

File: tests/raftstore/test_snap.rs [hunk 1]
```diff
@@ -136,10 +139,22 @@ fn test_snap_gc<T: Simulator>(cluster: &mut Cluster<T>) {
         cluster.must_put(b"k2", b"v2");
     }
 
+    let mut tried_cnt = 0;
+    loop {
+        let snap_index = rx.recv_timeout(Duration::from_secs(3)).unwrap();
+        if snap_index != first_snap_idx {
+            break;
+        }
+        tried_cnt += 0;
```
Comment (ngaut) [inline tests/raftstore/test_snap.rs]: what's this for ?

Comment (BusyJay) [inline tests/raftstore/test_snap.rs]: To make sure a different snapshot has been dropped.

Comment (siddontang) [inline tests/raftstore/test_snap.rs]: add 0 here?

Comment (BusyJay) [inline tests/raftstore/test_snap.rs]: My mistake.

File: tests/raftstore/test_snap.rs [hunk 2]
```diff
@@ -136,10 +139,21 @@ fn test_snap_gc<T: Simulator>(cluster: &mut Cluster<T>) {
         cluster.must_put(b"k2", b"v2");
     }
 
+    let mut timer = Instant::now();
```
Comment (ngaut) [inline tests/raftstore/test_snap.rs]: s/timer/now/

**Comment by ngaut** [comment]
LGTM

**Pull request merged**